src/clr-angular/layout/tabs/tabs.ts

```diff
@@ -76,12 +103,27 @@ export class ClrTabs implements AfterContentInit {
   }
 
   ngAfterContentInit() {
-    if (typeof this.ifActiveService.current === 'undefined') {
-      this.tabLinkDirectives.first.activate();
+    this._tabLinkDirectives = this.tabs.map(tab => tab.tabLink);
+    this.subscription = this.tabs.changes.subscribe(() => {
+      this._tabLinkDirectives = this.tabs.map(tab => tab.tabLink);
+    });
+
+    if (typeof this.ifActiveService.current === 'undefined' && this.tabLinkDirectives[0]) {
+      this.tabLinkDirectives[0].activate();
     }
   }
 
   toggleOverflow(event: any) {
     this.ifOpenService.toggleWithEvent(event);
   }
+
+  isVertical() {
+    return this.layout === TabsLayout.VERTICAL;
+  }
+
+  ngOnDestroy() {
+    if (this.subscription) {
```

Comment:
Our pattern has been to push subscriptions onto an array (even if there is only one for now, could always expand) and then here we just loop over each and unsubscribe. Saves this if block and expands more easily.

Reply:
Done. I just noticed we have both patterns in our code and decided to go for the short one. Good to know.